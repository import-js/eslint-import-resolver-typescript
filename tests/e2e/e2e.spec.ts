import fs from 'node:fs/promises'
import path from 'node:path'

import { tryFile } from '@pkgr/core'
import { exec } from 'tinyexec'

describe('e2e cases', async () => {
  const { dirname } = import.meta

  const dirents = await fs.readdir(dirname, { withFileTypes: true })

  for (const dirent of dirents) {
    const dirName = dirent.name

    if (!dirent.isDirectory() || dirName === '__snapshots__') {
      continue
    }

    const absoluteDir = path.resolve(dirname, dirName)

    if (tryFile('yarn.lock', false, absoluteDir)) {
      await exec('yarn', ['--immutable'], {
        nodeOptions: {
          cwd: absoluteDir,
        },
      })
    }

    it('should exec eslint successfully', async () => {
      const eslintConfig = tryFile('eslint.config.js', false, absoluteDir)
      expect(
        await exec(
          'yarn',
          [
            'eslint',
            ...(eslintConfig
              ? ['-c', eslintConfig]
              : ['--ignore-pattern', '!.dot']),
            '--ext',
            'cjs,cts,js,jsx,mjs,mts,ts,tsx',
            '--report-unused-disable-directives',
            absoluteDir,
          ],
          {
            nodeOptions: {
              env: {
                ESLINT_USE_FLAT_CONFIG: eslintConfig ? undefined : 'false',
                NODE_OPTIONS: '--no-warnings=ESLintRCWarning',
              },
            },
          },
        ),
      ).toMatchSnapshot(dirName)
    })
  }
})
