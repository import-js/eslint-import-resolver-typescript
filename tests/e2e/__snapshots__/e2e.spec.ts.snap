// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`e2e cases > should exec eslint successfully > absoluteAlias 1`] = `
{
  "exitCode": 0,
  "stderr": "",
  "stdout": "",
}
`;

exports[`e2e cases > should exec eslint successfully > dotInclude 1`] = `
{
  "exitCode": 0,
  "stderr": "",
  "stdout": "",
}
`;

exports[`e2e cases > should exec eslint successfully > dotPaths 1`] = `
{
  "exitCode": 0,
  "stderr": "",
  "stdout": "",
}
`;

exports[`e2e cases > should exec eslint successfully > dotProject 1`] = `
{
  "exitCode": 0,
  "stderr": "",
  "stdout": "",
}
`;

exports[`e2e cases > should exec eslint successfully > importXResolverV3 1`] = `
{
  "exitCode": 0,
  "stderr": "",
  "stdout": "",
}
`;

exports[`e2e cases > should exec eslint successfully > multipleEslintrcs 1`] = `
{
  "exitCode": 0,
  "stderr": "",
  "stdout": "",
}
`;

exports[`e2e cases > should exec eslint successfully > multipleTsconfigs 1`] = `
{
  "exitCode": 0,
  "stderr": "",
  "stdout": "",
}
`;

exports[`e2e cases > should exec eslint successfully > nearestTsconfig 1`] = `
{
  "exitCode": 0,
  "stderr": "",
  "stdout": "",
}
`;

exports[`e2e cases > should exec eslint successfully > withJsExtension 1`] = `
{
  "exitCode": 0,
  "stderr": "",
  "stdout": "",
}
`;

exports[`e2e cases > should exec eslint successfully > withJsconfig 1`] = `
{
  "exitCode": 0,
  "stderr": "",
  "stdout": "",
}
`;

exports[`e2e cases > should exec eslint successfully > withPaths 1`] = `
{
  "exitCode": 0,
  "stderr": "",
  "stdout": "",
}
`;

exports[`e2e cases > should exec eslint successfully > withPathsAndNestedBaseUrl 1`] = `
{
  "exitCode": 0,
  "stderr": "",
  "stdout": "",
}
`;

exports[`e2e cases > should exec eslint successfully > withQuerystring 1`] = `
{
  "exitCode": 0,
  "stderr": "",
  "stdout": "",
}
`;

exports[`e2e cases > should exec eslint successfully > withoutPaths 1`] = `
{
  "exitCode": 0,
  "stderr": "",
  "stdout": "",
}
`;
